packages/vite/src/node/plugins/resolve.ts

```diff
@@ -1040,52 +1039,8 @@ function packageEntryFailure(id: string, details?: string) {
   )
 }
 
-const conditionalConditions = new Set(['production', 'development', 'module'])
-
-function resolveExports(
-  pkg: PackageData['data'],
-  key: string,
-  options: InternalResolveOptionsWithOverrideConditions,
-  targetWeb: boolean
-) {
-  const overrideConditions = options.overrideConditions
-    ? new Set(options.overrideConditions)
-    : undefined
-
-  const conditions = []
-  if (
-    (!overrideConditions || overrideConditions.has('production')) &&
-    options.isProduction
-  ) {
-    conditions.push('production')
-  }
-  if (
-    (!overrideConditions || overrideConditions.has('development')) &&
-    !options.isProduction
-  ) {
-    conditions.push('development')
-  }
-  if (
-    (!overrideConditions || overrideConditions.has('module')) &&
-    !options.isRequire
-  ) {
-    conditions.push('module')
-  }
-  if (options.overrideConditions) {
-    conditions.push(
-      ...options.overrideConditions.filter((condition) =>
-        conditionalConditions.has(condition)
-      )
-    )
-  } else if (options.conditions.length > 0) {
-    conditions.push(...options.conditions)
-  }
-
-  return _resolveExports(pkg, key, {
-    browser: targetWeb && !conditions.includes('node'),
-    require: options.isRequire && !conditions.includes('import'),
-    conditions
-  })
+function getInlineConditions(conditions: string[], targetWeb: boolean) {
+  return targetWeb && !conditions.includes('node') ? ['browser'] : ['node']
```

Comment:
This mirrors the `browser` option of lukeed's `resolve.exports` module, but we *might* want to replace the `['node']` expression with `[]` to avoid the assumption that non-browser environment is using Node.js